src/test/java/net/snowflake/client/core/arrow/DateConverterTest.java

```diff
@@ -162,4 +165,76 @@ public void testRandomDates() throws SFException {
+  public void testTimezoneDates() throws SFException {
+    int testDay = 16911;
+    Map<String, String> customFieldMeta = new HashMap<>();
+    customFieldMeta.put("logicalType", "DATE");
+    // test normal date
+    FieldType fieldType =
+        new FieldType(true, Types.MinorType.DATEDAY.getType(), null, customFieldMeta);
+
+    DateDayVector vector = new DateDayVector("date", fieldType, allocator);
+
+    vector.setSafe(0, testDay);
+
+    // Test JDBC_FORMAT_DATE_WITH_TIMEZONE=TRUE with different session timezones
+    TimeZone.setDefault(TimeZone.getTimeZone("UTC"));
+    ArrowVectorConverter converter = new DateConverter(vector, 0, this, true);
+    converter.setUseSessionTimezone(true);
+    converter.setSessionTimeZone(TimeZone.getTimeZone(System.getProperty("user.timezone")));
+    Object obj = converter.toObject(0);
+    Object utcObj =
+        ArrowResultUtil.getDate(
+            testDay,
+            TimeZone.getTimeZone("UTC"),
+            TimeZone.getTimeZone(System.getProperty("user.timezone")));
+
+    String tz = System.getProperty("user.timezone");
+    switch (System.getProperty("user.timezone")) {
+      case ("UTC"):
+        assertThat(obj.toString(), is("2016-04-20"));
+        assertThat(utcObj.toString(), is("2016-04-20"));
+        assertThat(((Date) obj).getTime(), is(((Date) utcObj).getTime()));
+        break;
+      case ("America/Los_Angeles"):
+        assertThat(obj.toString(), is("2016-04-20"));
+        assertThat(utcObj.toString(), is("2016-04-20"));
+        assertThat(
+            ((Date) obj).getTime(), is(((Date) utcObj).getTime() - (7 * MILLIS_IN_ONE_HOUR)));
+        break;
+      case ("America/New_York"):
+        assertThat(obj.toString(), is("2016-04-20"));
+        assertThat(utcObj.toString(), is("2016-04-20"));
+        assertThat(
+            ((Date) obj).getTime(), is(((Date) utcObj).getTime() - (4 * MILLIS_IN_ONE_HOUR)));
+        break;
+      case ("Pacific/Honolulu"):
+        assertThat(obj.toString(), is("2016-04-20"));
+        assertThat(utcObj.toString(), is("2016-04-20"));
+        assertThat(
+            ((Date) obj).getTime(), is(((Date) utcObj).getTime() - (10 * MILLIS_IN_ONE_HOUR)));
+        break;
+      case ("Asia/Singapore"):
+        assertThat(obj.toString(), is("2016-04-20"));
+        assertThat(utcObj.toString(), is("2016-04-19"));
+        assertThat(
+            ((Date) obj).getTime(), is(((Date) utcObj).getTime() + (8 * MILLIS_IN_ONE_HOUR)));
+        break;
+      case ("MEZ"):
+        assertThat(obj.toString(), is("2016-04-20"));
+        assertThat(utcObj.toString(), is("2016-04-20"));
+        assertThat(((Date) obj).getTime(), is(((Date) utcObj).getTime()));
+        break;
+      case ("MESZ"):
+        assertThat(obj.toString(), is("2016-04-20"));
+        assertThat(utcObj.toString(), is("2016-04-20"));
+        assertThat(((Date) obj).getTime(), is(((Date) utcObj).getTime()));
+        break;
+      default:
+        fail("Unknown timezone");
```

Comment:
do we really want to fail for other timezones? we should handle this case

Reply:
My intention was if other timezones are added to test, this should be updated. But I can instead just change it to write a comment to console